CHAID/CHAID.py

```diff
@@ -396,16 +407,20 @@ def generate_best_split(self, ind, dep):
                         [col2_freq.get(k, 0) for k in keys]
                     ]
 
-                    chi = stats.chi2_contingency(np.array(cr_table), correction=False)
-                    sub_data[j] = (comb, chi[0], chi[1])
+                    ret = chi2(np.array(cr_table), correction=False)
+                    sub_data[j] = (comb, ret[1], ret[0])
 
-                choice, chi, highest_p_split = max(sub_data, key=lambda x: (x[2], x[1]))
+                choice, highest_p_join, _ = max(sub_data, key=lambda x: (x[1], x[2]))
 
-                if highest_p_split < self.alpha_merge:
+                if highest_p_join < self.alpha_merge:
+                    cr_table = [
```

Comment:
At the time we made that choice it was a good call, now I think it might be too much. I'm hoping we can factor out a lot of the frequency code into CHAIDVector for variable types. But yeah. Let's talk when I'm bacj